src/Mvc/Extensions.ApiDescription.Client/src/build/Microsoft.Extensions.ApiDescription.Client.targets

```
@@ -0,0 +1,151 @@
+﻿<?xml version="1.0" encoding="utf-8" standalone="no"?>
+<Project>
+  <!-- Internal settings. Not intended for customization. -->
+  <PropertyGroup>
+    <GenerateOpenApiReferenceCodeDependsOn>
+      _GetMetadataForOpenApiReferences;
+      _GenerateOpenApiReferenceCode;
+      _CreateCompileItemsForOpenApiReferences
+    </GenerateOpenApiReferenceCodeDependsOn>
+    <GenerateServiceFileReferenceCodeDependsOn>
+      _GenerateErrorsForOldItems;
+      _CreateOpenApiReferenceItemsForOpenApiProjectReferences;
+      GenerateOpenApiReferenceCode
+    </GenerateServiceFileReferenceCodeDependsOn>
+  </PropertyGroup>
+
+  <!-- OpenApiProjectReference support. -->
+
+  <ItemGroup>
+    <!--
+      Do not change %(ReferenceOutputAssembly) if project contains duplicate @(ProjectReference) and
+      @(OpenApiProjectReference) items.
+    -->
+    <ProjectReference Include="@(OpenApiProjectReference)" Exclude="@(ProjectReference)" ReferenceOutputAssembly="false" />
+    <ProjectReference Update="@(OpenApiProjectReference)" OpenApiReference="true" />
+  </ItemGroup>
+
+  <Target Name="_CreateOpenApiReferenceItemsForOpenApiProjectReferences"
+      AfterTargets="ResolveProjectReferences"
+      Condition="'%(ProjectReferenceWithConfiguration.OpenApiReference)' == 'true' AND '$(DesignTimeBuild)' != 'true' AND '$(BuildingProject)' == 'true'"
+      Inputs="%(ProjectReferenceWithConfiguration.Identity)"
+      Outputs="&lt;not-a-file !&gt;">
+    <PropertyGroup>
+      <_Identity>%(ProjectReferenceWithConfiguration.Identity)</_Identity>
+      <_Temporary />
+    </PropertyGroup>
+
+    <MSBuild Target="OpenApiGetDocuments"
+        Project="$(_Identity)"
+        Properties="%(ProjectReferenceWithConfiguration.SetConfiguration); %(ProjectReferenceWithConfiguration.SetPlatform); %(ProjectReferenceWithConfiguration.SetTargetFramework)"
+        RebaseOutputs="true"
+        RemoveProperties="%(ProjectReferenceWithConfiguration.GlobalPropertiesToRemove);TargetFrameworks;RuntimeIdentifier">
+      <Output TaskParameter="TargetOutputs" PropertyName="_Temporary" />
```

There are some MSBuild way of doing this which don't require manually matching up some returned metadata from  a target. I would start by changing this to return an item instead of a property.

```suggestion
      <Output TaskParameter="TargetOutputs" ItemName="_ResolvedOpenApiProjectWithDocuments" />
```

Next, make the target you're invoking, in this case "OpenApiGetDocuments", look like this:

```xml
<Target Name="OpenApiGetDocuments"
              Returns="@(OpenApiProjectWithDocuments)">
   <ItemGroup>
       <OpenApiProjectWithDocuments Include="$(MSBuildProjectFullPath)">
          <Documents>$(stuffHere)</Documents>
      </OpenApiProjectWithDocuments>
  </ItemGroup>
</Target>
```